libcudacxx/include/cuda/__ptx/instructions/cp_async_bulk_wait_group.h

```diff
@@ -41,13 +41,13 @@ __device__ static inline void cp_async_bulk_wait_group(
 #if __cccl_ptx_isa >= 800
 extern "C" _CCCL_DEVICE void __cuda_ptx_cp_async_bulk_wait_group_is_not_supported_before_SM_90__();
 template <int _N32>
-_CCCL_DEVICE static inline void cp_async_bulk_wait_group(n32_t<_N32> __N)
+_CCCL_DEVICE static inline void cp_async_bulk_wait_group(n32_t<_N32> __n)
 {
   NV_IF_ELSE_TARGET(
     NV_PROVIDES_SM_90,
     (asm volatile("cp.async.bulk.wait_group %0;"
                   :
-                  : "n"(__N)
+                  : "n"(__n.value)
```

Comment:
Suggestion: How about using just the template parameter `_N32`? I am not certain `__n` is a constant expression here, and therefore some compilers complain that `__n.value` isn't either. I am also not sure whether that's required here.

Reply:
`integral_constant.value` is a static constexpr variable, so `__n.value` is a constant expression even if it is accessed through a non-constant value

Reply:
I remember running into issues with this at some point, but I just checked on compiler explorer and could not reproduce it. All good then!

Reply:
This popped up during one of the compilation tests I ran trying to import v2.5.0.  Does the error get triggered on one of cccl's tests w/ clang in your CI? I assume it should. It would give the definitive answer whether clang is OK with `__n.value`.

Reply:
Looks like the tests do not build for sm_90a, so they would not catch stuff hidden behind `NV_HAS_FEATURE_SM_90a,`:
```
[41/1786] /usr/bin/sccache /usr/bin/clang++  -Dheadertest___ptx_instructions_tensormap_cp_fenceproxy_h_EXPORTS -I/home/coder/cccl/libcudacxx/include -O3 -DNDEBUG -std=gnu++20 --cuda-gpu-arch=sm_60 --cuda-gpu-arch=sm_70 --cuda-gpu-arch=sm_80 
```

Considering that there's nontrivial amount of interesting functionality that's specific to sm_90 and sm_90a, it may be useful to build/test for them, too.

Reply:
That is really strange, considering we especially have a target for SM90a for that reason

Reply:
Oh you know what, our SM90a tests only runs on nvcc with gcc https://github.com/NVIDIA/cccl/actions/runs/10368241031/job/28701520566

Reply:
I added two CI runs for SM90 and SM90a, lets see what happens with that